plugins/inputs/cisco_telemetry_mdt_netconf/cisco_telemetry_mdt_netconf.go

```diff
@@ -0,0 +1,941 @@
+type dialinSubscriptionRequest struct {
+	XPathFilter   string `toml:"xpath_filter"`
+	UpdateTrigger string `toml:"update_trigger"`
+	Period        config.Duration
+	Keys          []string
+}
+
+// notificationSubscriptionRequest given as input to the plugin
+type notificationSubscriptionRequest struct {
+	Stream string `toml:"stream"`
+	Keys   []string
+}
+
+// getRequest given as input to the plugin
+type getRequest struct {
+	SelectFilter string `toml:"xpath_filter"`
+	Period       config.Duration
+	Keys         []string
+}
+
+// netconfSubscriptionRequest sent through NETCONF, as per specification
+type netconfSubscriptionRequest struct {
+	XMLName         xml.Name `xml:"urn:ietf:params:xml:ns:yang:ietf-event-notifications establish-subscription"`
+	YangPush        string   `xml:"xmlns:yp,attr"`
+	Stream          string   `xml:"stream"`
+	XPathFilter     string   `xml:"urn:ietf:params:xml:ns:yang:ietf-yang-push xpath-filter"`
+	Period          uint64   `xml:"urn:ietf:params:xml:ns:yang:ietf-yang-push period,omitempty"`           // hundreds of a second 100=1s
+	DampeningPeriod *uint64  `xml:"urn:ietf:params:xml:ns:yang:ietf-yang-push dampening-period,omitempty"` // on-change
+}
+
+// netconfYangPush notification received through NETCONF, as per specification
+type netconfYangPush struct {
+	netconf.Notification
+	PushUpdate struct {
+		Content struct {
+			InnerXML []byte `xml:",innerxml"`
+		} `xml:"datastore-contents-xml"`
+	} `xml:"urn:ietf:params:xml:ns:yang:ietf-yang-push push-update"`
+}
+
+// netconfYangEvent notification received through NETCONF, as per specification
+type netconfYangEvent struct {
+	netconf.Notification
+	InnerXML []byte `xml:",innerxml"`
+}
+
+// CiscoTelemetryNETCONF plugin
+type CiscoTelemetryNETCONF struct {
+	ServerAddress string `toml:"server_address"`
+
+	Dsrs *dialinSubscriptionRequestsService `toml:"subscription_service"`
+	Grs  *getRequestsService                `toml:"get_service"`
+
+	// SSH connection - credentials
+	Username string
+	Password string
+
+	// SSH connection - security
+	IgnoreServerAuthenticity bool
+	ServerPublicKey          string
+
+	// Xpaths and keys of all operations
+	userXpaths map[string]interface{}
+	userKeys   map[string]interface{}
+
+	Redial config.Duration
+
+	acc    telegraf.Accumulator
+	cancel context.CancelFunc
+
+	// Waitgroup for monitoring the main services
+	waitgroup sync.WaitGroup
+}
+
+// Init ensures the plugin is configured correctly.
+func (c *CiscoTelemetryNETCONF) Init() error {
+	if c.ServerAddress == "" {
+		return fmt.Errorf("server address cannot be empty")
+	}
+	if c.Username == "" || c.Password == "" {
+		return fmt.Errorf("ssh credentials cannot be empty")
+	}
+	if !c.IgnoreServerAuthenticity && c.ServerPublicKey == "" {
+		return fmt.Errorf("public key must exist when ignore server authenticity is enabled")
+	}
+	if c.Redial < 1 {
+		return fmt.Errorf("redial must be greater than or equal to 1")
```

Comment:
Greater than '1'? Do you mean 1 second, minute, ...?

Reply:
I've rewritten the check. We check if it is greater than 1s. Changes are here: https://github.com/cisco-ie/telegraf/commit/018fdd7514f7326ba049d0886d647bd69467e07e. Thank you.